### PR TITLE
Glossary View OMAS - referenceable entity retrieval, fixed param validation, simplified conversion

### PR DESCRIPTION
**Pull request by lcpopa**
Signed-off-by: Liviu Popa <liviu-constantin.popa@ing.com>

### CODE REVIEW

**Comment by planetf1** [comment]
recheck

File: open-metadata-implementation/access-services/glossary-view/glossary-view-spring/pom.xml [hunk 1]
```diff
@@ -54,9 +54,9 @@
         </dependency>
 
         <dependency>
-            <groupId>javax.validation</groupId>
-            <artifactId>validation-api</artifactId>
-            <version>2.0.1.Final</version>
+            <groupId>org.springframework.boot</groupId>
```
Comment (planetf1) [inline open-metadata-implementation/access-services/glossary-view/glossary-view-spring/pom.xml]: As this is a new dependency you'll now need to add to the top level pom an entry IN THE DEPENDENCYMANAGEMENT section  like
```
          <dependency>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-starter-validation</artifactId>
                <scope>compile</scope>
                <version>${spring-boot.version}</version>
            </dependency>
```
Then in this pom do not include the version tag

File: open-metadata-implementation/access-services/glossary-view/glossary-view-spring/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/server/spring/TermResource.java [hunk 2]
```diff
@@ -384,4 +388,31 @@ public GlossaryViewEntityDetailResponse getUsedInContexts(@PathVariable("serverN
...
+     * @param termGUID term GUID
+     * @param from from
+     * @param size size
+     *
+     * @return valid values
+     */
+    @RequestMapping(method = RequestMethod.GET, path = "/terms/{termGUID}/assigned-elements")
+    public GlossaryViewEntityDetailResponse getAssignedElements(@PathVariable("serverName") String serverName,
+                                                              @PathVariable("userId") String userId,
+                                                              @PathVariable("termGUID") String termGUID,
```
Comment (ruxandraRosu) [inline open-metadata-implementation/access-services/glossary-view/glossary-view-spring/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/server/spring/TermResource.java]: I think you can add a validation for non-blank termGUID

File: open-metadata-implementation/access-services/glossary-view/glossary-view-server/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/server/service/TermService.java [hunk 3]
```diff
@@ -50,6 +50,9 @@
     private static final String USED_IN_CONTEXT_RELATIONSHIP_NAME = "UsedInContext";
     private static final String USED_IN_CONTEXT_RELATIONSHIP_GUID = "2dc524d2-e29f-4186-9081-72ea956c75de";
 
+    private static final String SEMANTIC_ASSIGNMENT_RELATIONSHIP_NAME = "SemanticAssignment";
+    private static final String SEMANTIC_ASSIGNMENT_RELATIONSHIP_GUID = "e6670973-645f-441a-bec7-6f5570345b92";
```
Comment (ruxandraRosu) [inline open-metadata-implementation/access-services/glossary-view/glossary-view-server/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/server/service/TermService.java]: I think you can use the helper to load the types by name in order not to hardcode both type name and type guid.

File: open-metadata-implementation/access-services/glossary-view/glossary-view-spring/pom.xml (hunk 1 above)
Comment (lcpopa) [reply open-metadata-implementation/access-services/glossary-view/glossary-view-spring/pom.xml]: Done. Thank you for the explanation

**Comment by lcpopa** [comment]
Closing as commits from other contributors are displayed. Will reopen

**Pull request closed without merging**